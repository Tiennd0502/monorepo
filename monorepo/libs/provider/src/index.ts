export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
