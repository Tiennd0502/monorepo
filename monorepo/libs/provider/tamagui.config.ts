import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
