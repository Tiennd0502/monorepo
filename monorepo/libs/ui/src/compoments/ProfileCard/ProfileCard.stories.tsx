import { Meta, StoryObj } from '@storybook/react';
import { Stack } from 'tamagui';

import ProfileCard from '.';

const meta: Meta<typeof ProfileCard> = {
  title: 'Components/ProfileCard',
  component: ProfileCard,
  decorators: [
    (Story) => (
      <Stack width={500}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof ProfileCard>;

export const Default: Story = {
  args: {
    title: 'My Orders',
    description: 'Already have 10 orders',
    onPress: () => null,
  },
};
