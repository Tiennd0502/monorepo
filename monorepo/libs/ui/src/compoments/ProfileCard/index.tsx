import { memo } from 'react';
import { Stack, XStack, StackProps } from 'tamagui';

import Text from '../Text';
import { shadows } from '../../themes';
import { ChevronRightIcon } from '../icons';

interface ProfileCardProps extends StackProps {
  title: string;
  description?: string;
  onPress: () => void;
}

const ProfileCard = ({
  title,
  description = '',
  onPress,
  style,
  ...props
}: ProfileCardProps) => (
  <XStack
    testID='profile-card'
    paddingVertical="$5"
    paddingLeft="$5"
    paddingRight="$2.5"
    backgroundColor="$secondary"
    alignItems="center"
    justifyContent="space-between"
    style={[shadows.card, style]}
    onPress={onPress}
    {...props}
  >
    <Stack gap="$1.5">
      <Text bold color="$primary" size="large">
        {title}
      </Text>
      {description && (
        <Text color="$textLabel" fontSize="$3">
          {description}
        </Text>
      )}
    </Stack>
    <ChevronRightIcon />
  </XStack>
);

export default memo(ProfileCard);
