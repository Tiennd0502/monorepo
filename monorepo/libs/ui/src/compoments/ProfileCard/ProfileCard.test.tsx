import { fireEvent, render, waitFor } from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import ProfileCard from '.';

const onPress = jest.fn();

const baseProps = {
	title: 'My Orders',
	description: 'Already have 10 orders',
	onPress,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <ProfileCard {...props} />
    </TamaguiProvider>
  );
};


describe('ProfileCard', () => {

  it('should render ProfileCard component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('calls onPress when item clicked', () => {
    const {getByTestId} = setup();
    waitFor(() => {
      const editIcon = getByTestId('profile-card');
      fireEvent.press(editIcon);

      expect(onPress).toHaveBeenCalled();
    });
  });
});
