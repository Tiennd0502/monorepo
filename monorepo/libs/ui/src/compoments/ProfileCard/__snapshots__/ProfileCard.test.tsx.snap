// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard should render ProfileCard component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "elevation": 16,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 10,
      "paddingTop": 20,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
  testID="profile-card"
>
  <View
    style={
      {
        "gap": 6,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Merriweather, Helvetica, Arial, sans-serif",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      My Orders
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 12,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Already have 10 orders
    </Text>
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    fill="none"
    focusable={false}
    height="24"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="24"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M9 18l6-6-6-6"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4280558628,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="1.5"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
